wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/convert/component/plugin/PluginComponentToWorkspaceApplier.java

```diff
@@ -73,25 +73,27 @@ public void apply(
 
     String workspacePluginsAttribute =
         workspaceConfig.getAttributes().get(WORKSPACE_TOOLING_PLUGINS_ATTRIBUTE);
-    workspaceConfig
-        .getAttributes()
-        .put(
-            WORKSPACE_TOOLING_PLUGINS_ATTRIBUTE,
-            append(workspacePluginsAttribute, pluginComponent.getId()));
 
-    String pluginsAliases =
-        workspaceConfig.getAttributes().get(PLUGINS_COMPONENTS_ALIASES_WORKSPACE_ATTRIBUTE);
-    if (pluginComponent.getAlias() != null) {
-      workspaceConfig
-          .getAttributes()
-          .put(
-              PLUGINS_COMPONENTS_ALIASES_WORKSPACE_ATTRIBUTE,
-              append(pluginsAliases, pluginComponent.getId() + "=" + pluginComponent.getAlias()));
-    }
+    final String reference = pluginComponent.getReference();
+    final String pluginId = pluginComponent.getId();
+    final String registryUrl = pluginComponent.getRegistryUrl();
+    final String compositeId = registryUrl != null ? registryUrl + "#" + pluginId : pluginId;
```

Comment:
We're doing this composition on 2 different places. Maybe it would be better to abstract that into some utility method to define the logic and format only once?

Reply:
I agree. For now i made it via format in constants. About moving to common method agree too, but it is require utility or abstract class since those places are in different packages. OTOH we plan to switch to json format soon, so maybe it is overkill  to create a new class for this operaton now.

Reply:
Also, I see the following code
```java
    final ExtendedPluginFQN fqn;
    try {
      if (!isNullOrEmpty(reference)) {
        fqn = fqnParser.evaluateFqn(reference, contentProvider);
      } else {
        fqn = fqnParser.parsePluginFQN(compositeId);
      }
    } catch (InfrastructureException e) {
      throw new DevfileException(e.getMessage(), e);
    }
```
in three places and it makes sense to have utility class/method for 
```java
    final ExtendedPluginFQN fqn = componentFqnParser.evaluate(component, contentProvider);
```

about 
```
    final String compositeId = registryUrl != null ? registryUrl + "#" + pluginId : pluginId;
```
I see this method as `getReference` in `ExtendedPluginFQN`, but I'm OK to leave as is since we have a plan to use JSON instead of reference.